trade-finance-manager-api/src/v1/controllers/user/user.routes.js

```diff
@@ -60,6 +65,31 @@ module.exports.createTfmUser = (req, res, next) => {
   });
 };
 
+module.exports.upsertTfmUserFromEntraIdUser = async (req, res, next) => {
+  try {
+    const entraIdUser = ENTRA_ID_USER_SCHEMA.parse(req.body);
+    const tfmUser = await upsertTfmUserFromEntraIdUser({ entraIdUser, auditDetails: generateSystemAuditDetails() });
+
+    return res.status(200).send(tfmUser);
+  } catch (error) {
+    if (error instanceof ApiError) {
+      return res.status(error.status).send({
+        status: error.status,
+        message: error.message,
+        code: error.code,
+      });
+    }
+    if (error instanceof ZodError) {
+      return res.status(HttpStatusCode.BadRequest).send({
+        status: HttpStatusCode.BadRequest,
+        message: 'Error validating payload',
+      });
+    }
+
+    return next(error);
```

Comment:
Should we be returning `next` if there is an unknown error? Maybe return 400/500 or throw a generic error instead?

Reply:
It'll get handled by the built in middleware and return a 500 automatigically :)